cpp/arcticdb/util/reliable_storage_lock.tpp

```diff
@@ -0,0 +1,242 @@
+/* Copyright 2023 Man Group Operations Limited
+ *
+ * Use of this software is governed by the Business Source License 1.1 included in the file licenses/BSL.txt.
+ *
+ * As of the Change Date specified in that file, in accordance with the Business Source License, use of this software will be governed by the Apache License, version 2.0.
+ */
+
+#include <arcticdb/util/reliable_storage_lock.hpp>
+
+#include <arcticdb/storage/store.hpp>
+#include <arcticdb/entity/types.hpp>
+#include <arcticdb/util/constants.hpp>
+#include <arcticdb/util/configs_map.hpp>
+
+#include <mutex>
+#include <folly/experimental/FunctionScheduler.h>
+
+namespace arcticdb {
+
+namespace lock {
+
+const auto SEPARATOR = '*';
+const auto EXTENDS_PER_TIMEOUT = 5u;
+
+inline StreamDescriptor lock_stream_descriptor(const StreamId &stream_id) {
+    return StreamDescriptor{stream_descriptor(
```

Comment:
I don't think it needs the other constructor as the function already returns a stream descriptor